httpbin/core.py

```diff
@@ -84,6 +84,30 @@ def jsonify(*args, **kwargs):
 # -----------
 # Middlewares
 # -----------
+"""
+https://github.com/kennethreitz/httpbin/issues/340
+Adds a middleware to provide chunked request encoding support running under
+gunicorn only.
+Werkzeug required environ 'wsgi.input_terminated' to be set otherwise it
+empties the input request stream.
+- gunicorn seems to support input_terminated but does not add the environ,
+  so we add it here.
+- flask will hang and does not seem to properly terminate the request, so
+  we explicityly deny chunked requests.
+"""
+@app.before_request
+def before_request():
+    if request.environ.get('HTTP_TRANSFER_ENCODING', '').lower() == 'chunked':
+        server = request.environ.get('SERVER_SOFTWARE', '')
+        if server.lower().startswith('gunicorn/'):
+            if 'wsgi.input_terminated' in request.environ:
+                app.logger.warning("environ wsgi.input_terminated already set, keeping: %s"
```

Comment:
Is this really a warning level message? I'd consider this `INFO` or `DEBUG` at best.

Reply:
Yes `debug` is better and I have changed to that and rebased to master.

The original thinking of a louder log message was to quickly catch a change in `gunicorn` if it started adding `input_terminated` to requests, deprecating this middleware.  But debug is fine for this.